lighthouse-core/audits/seo/tap-targets.js

```diff
@@ -34,130 +50,108 @@ function clientRectBelowMinimumSize(cr) {
 
 /**
  * A target is "too small" if none of its clientRects are at least the size of a finger.
- * @param {LH.Artifacts.TapTarget[]} targets
- * @returns {LH.Artifacts.TapTarget[]}
+ * @param {BoundedTapTarget[]} targets
+ * @returns {BoundedTapTarget[]}
  */
 function getTooSmallTargets(targets) {
   return targets.filter(target => {
-    return target.clientRects.every(clientRectBelowMinimumSize);
+    return target.tapTarget.clientRects.every(clientRectBelowMinimumSize);
   });
 }
 
 /**
- * @param {LH.Artifacts.TapTarget[]} tooSmallTargets
- * @param {LH.Artifacts.TapTarget[]} allTargets
+ * @param {BoundedTapTarget[]} tooSmallTargets
+ * @param {BoundedTapTarget[]} allTargets
  * @returns {TapTargetOverlapFailure[]}
  */
 function getAllOverlapFailures(tooSmallTargets, allTargets) {
   /** @type {TapTargetOverlapFailure[]} */
-  let failures = [];
+  const failures = [];
 
   tooSmallTargets.forEach(target => {
-    const overlappingTargets = getAllOverlapFailuresForTarget(
-      target,
-      allTargets
-    );
+    // Convert client rects to unique tappable areas from a user's perspective
+    const tappableRects = getTappableRectsFromClientRects(target.tapTarget.clientRects);
 
-    failures = failures.concat(overlappingTargets);
-  });
+    for (const maybeOverlappingTarget of allTargets) {
+      if (maybeOverlappingTarget === target) {
+        // Checking the same target with itself, skip.
+        continue;
+      }
 
-  return failures;
-}
+      if (!rectsTouchOrOverlap(target.boundingRect, maybeOverlappingTarget.boundingRect)) {
+        // Bounding boxes (padded with half FINGER_SIZE_PX) don't overlap, skip.
```

Comment:
nit: should the prop be called `paddedBoundingRect`? seems like it might eliminate the need for part of this comment

Reply:
> nit: should the prop be called paddedBoundingRect

👍